.github/workflows/main.yml

```diff
@@ -0,0 +1,240 @@
+          - artifact-name: Win64
+            architecture: x64
+            build-options: "-PskipRelease -PskipSources"
+          - artifact-name: Win32
+            architecture: x86
+            build-options: "-PskipRelease -PskipSources"
+          - artifact-name: WinArm64
+            architecture: x64
+            build-options: "-PskipRelease -PskipSources -Pplatform=windows-arm64"
+
+    name: "Build - ${{ matrix.artifact-name }}"
+    runs-on: windows-2019
+    steps:
+      - uses: actions/checkout@v4
+        with:
+          submodules: 'true'
+      - uses: actions/setup-java@v4
+        with:
+          java-version: 11
+          distribution: 'zulu'
+          architecture: ${{ matrix.architecture }}
+      - name: Set Java Heap Size
+        run: sed -i 's/-Xmx2g/-Xmx1g/g' gradle.properties
+        if: matrix.architecture == 'x86'
+      - uses: lukka/run-vcpkg@v11.5
+        with:
+          runVcpkgInstall: true
+          vcpkgGitCommitId: '29b2ea2d4b6197e66ef346e62ccbba35b55b7de5'
+      - run: vcpkg install
+        name: install packages
+      - run: |
+          cd vcpkg_installed
+          ls
+        name: show
+      - run: |
+          @call "C:\\Program Files (x86)\\Microsoft Visual Studio\\2019\\Enterprise\\VC\\Auxiliary\\Build\\vcvars64.bat"
+          set PATH=%PATH%;%ANT_HOME%\bin
+          gradlew publish ${{ matrix.build-options }}
+        shell: cmd
+        name: Build with Gradle
+      - uses: actions/upload-artifact@v4
+        with:
+          name: ${{ matrix.artifact-name }}
+          path: gradleDir/outputs/
+
+  build-mac:
+    name: "Build - macOS"
+    runs-on: macOS-14
+    steps:
+      - uses: actions/checkout@v4
+        with:
+          submodules: 'true'
+      - uses: actions/setup-java@v4
+        with:
+          java-version: 11
+          distribution: 'zulu'
+      - run: brew install cmake ninja
+        name: install ninja
+      - uses: lukka/run-vcpkg@v11.5
+        with:
+            runVcpkgInstall: true
+            vcpkgGitCommitId: '29b2ea2d4b6197e66ef346e62ccbba35b55b7de5'
+      - run: vcpkg install
+        name: install packages
+      - run: ./gradlew publish
+        name: Build with Gradle
+      - uses: actions/upload-artifact@v4
+        with:
+          name: macOS
+          path: gradleDir/outputs/
+
+  build-mac-arm:
+    name: "Build - macOS (Arm)"
+    runs-on: macOS-14
+    steps:
+      - uses: actions/checkout@v4
+        with:
+          submodules: 'true'
+      - uses: actions/setup-java@v4
+        with:
+          java-version: 11
+          distribution: 'zulu'
+      - run: brew install cmake ninja
+        name: install ninja
+      - uses: lukka/run-vcpkg@v11.5
+        with:
+            runVcpkgInstall: true
+            vcpkgGitCommitId: '29b2ea2d4b6197e66ef346e62ccbba35b55b7de5'
+      - run: vcpkg install
+        name: install packages
+      - run: ./gradlew publish -Pforcealternatemacbuild
```

Comment:
`-Pforcealternatemacbuild` doesn't actually do anything here and can be removed. Since that makes it no different than the other Mac job, something else needs to be done to make Mac builds work.